### PR TITLE
Fixes for ads and song quality

### PR DESCRIPTION
**Pull request by archit**
1. Added support for picking best song quality available url.
2. Added support for skipping ads in the station song stream.

### CODE REVIEW

**Comment by nixme** [comment]
ooooh. first pull request for this repo. :metal:

File: lib/pandora/song.rb [hunk 1]
```diff
@@ -59,7 +61,13 @@ def load_from_data(data, audio_formats)
       @album_explorer_url = data['albumExplorerUrl']
       @artist_explorer_url = data['artistDetailUrl']
 
-      @audio_url = data['audioUrl']
+      # Automagically pick best quality available.
+      i = 0
+      until @audio_url || i == ALL_AUDIO_QUALITIES.length
+        @audio_url = data['audioUrlMap']["#{ALL_AUDIO_QUALITIES[i]}Quality"]['audioUrl']
+        i += 1
+      end
```
Comment (nixme) [inline lib/pandora/song.rb:67]:
``` ruby
@audio_url = ALL_AUDIO_QUALITIES.find do |quality|
  data['audioUrlMap']["#{quality}Quality"]['audioUrl']
end
```

File: lib/pandora/station.rb [hunk 2]
```diff
@@ -47,8 +47,9 @@ def next_songs(audio_formats = Song::DEFAULT_AUDIO_FORMATS)
       }
 
       result['items'].map do |song_data|
+        next unless song_data['adToken'].nil?
```
Comment (nixme) [inline lib/pandora/station.rb]: `next if song_data['adToken']`

**Comment by archit** [comment]
Ready for review.

**Comment by nixme** [comment]
sweet! thanks

**Pull request merged**